### PR TITLE
support cancel building job

### PR DESCRIPTION
**Pull request by xiaomuqiao**
make changes to support cancel building job manually
Also fix issue https://github.com/caicloud/cyclone/issues/83

### CODE REVIEW

File: event/manager.go [hunk 1]
```diff
@@ -158,6 +158,17 @@ func eventChangeHandler(event *api.Event, preEvent *api.Event) {
 			log.Errorf("delete finished event err: %v", err)
 		}
 	}
+
+	// make the state of version as running, the frontend will be based on this state to decide whether to show cancel button.
+	if preEvent.Status == api.EventStatusPending && event.Status == api.EventStatusRunning && event.Operation == CreateVersionOps {
```
Comment (superxi911) [inline event/manager.go]: I think the event should can be canceled after it accepted.

Comment (xiaomuqiao) [inline event/manager.go]: but now  the front end is based on the version status to determine whether to display the cancel button

Comment (superxi911) [inline event/manager.go]: You can set the status of version to  "api.VersionRunning" when the event create.

Comment (xiaomuqiao) [inline event/manager.go]: ok

**Comment by superxi911** [comment]
LGTM

**Pull request merged**